.github/workflows/main.yml

```
@@ -1,98 +1,90 @@
+          path: test-results.xml
+          retention-days: 30
+      - name: Upload coverage
+        if: ${{ !cancelled() }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-${{ matrix.os }}-py${{ matrix.python-version }}
+          path: coverage.xml
+          retention-days: 30
+      - name: Test summary
+        if: ${{ !cancelled() && runner.os == 'Linux' }}
+        run: |
```

actionlint (1.7.12)

[error] shellcheck reported issue in this script: SC2086:info:1:84: Double quote to prevent globbing and word splitting

(shellcheck)